crates/credentials/src/pex/presentation_definition.rs

```diff
@@ -0,0 +1,292 @@
+/// Input Descriptors are used to describe the information a Verifier requires of a Holder.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InputDescriptor {
+    pub id: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub purpose: Option<String>,
+    pub constraints: Constraints,
+}
+
+/// Contains the requirements for a given Input Descriptor.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Constraints {
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub fields: Vec<Field>,
+}
+
+/// Contains the requirements for a given field within a proof.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Field {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub id: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub path: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub purpose: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub filter: Option<Filter>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub optional: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub predicate: Option<Optionality>,
+}
+
+/// Type alias for the possible values of the predicate field.
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Optionality {
+    Required,
+    Preferred,
+}
+
+/// A JSON Schema that is applied against the value of a field.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Filter {
+    #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
+    pub r#type: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub pattern: Option<String>,
+    #[serde(rename = "const", skip_serializing_if = "Option::is_none")]
+    pub const_value: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub contains: Option<Box<Filter>>,
+}
+
+fn generate_token() -> String {
+    let mut rng = rand::thread_rng();
+    let bytes: Vec<u8> = (0..16).map(|_| rng.gen_range(0..256) as u8).collect();
+    hex::encode(bytes)
+}
+
+impl PresentationDefinition {
+    pub fn select_credentials(&self, vc_jwts: &Vec<String>) -> Result<Vec<String>> {
+        let mut matches: HashSet<String> = HashSet::new();
+
+        for input_descriptor in &self.input_descriptors {
+            let matching_vc_jwts = input_descriptor.select_credentials(vc_jwts)?;
+            if matching_vc_jwts.is_empty() {
+                return Ok(vec![]);
+            }
+            matches.extend(matching_vc_jwts);
+        }
+
+        Ok(matches.into_iter().collect())
+    }
+}
+
+struct TokenizedField<'a> {
+    pub token: String,
+    pub path: &'a String,
+}
+
+impl InputDescriptor {
```

Comment:
in the other libs we have 

PresentationExchange.select_credentials
PresentationExchange.createPresentationFromCredentials

...

Was the plan to keep it similar or are there other api design changes in the pipeline?

https://github.com/TBD54566975/web5-js/tree/main/packages/credentials#presentationexchange

Reply:
The priority for now (keep me honest @frankhinek @mistermoe) is only `select_credentials` since that is needed for TBDex but `createPresentationFromCredentials` is not. Instead of using Presentation Submissions in TBDex RFQ messages, we submit a list of claims in the `Rfq.data.required_claims` field.